### ISSUE TITLE
Temp files not removed from Windows temp directory.

### ISSUE THREAD

**Issue by michealhall** [lovell/sharp#1503]
I'm integrating sharp into my project. Thanks for this library! It's very easy to use.

My initial implementation is very simple - a rotation using EXIF data and a resize and save. Currently running on a Windows 10 machine.

My issue is that I've found that certain values of fit result in temp files left behind (50GB worth in one test last night, which is how I discovered the problem!).  The particular fit options that display the problem are: fill, cover and outside.

I saw mentions for what looked like similar issues (#415), but I tried using sharp.cache(false) with no success.

If I use fit options of 'inside' or 'contains', there is no issue and the temp files are deleted. 

The function I'm using is a simple one:

``` javascript
 function resizeSharp( inputPath, outputPath, format, rotate, width, height ) {

    sharp.cache(false);

    const s = sharp( inputPath );

    s.withMetadata();

    ( rotate ) &&
      s.rotate();

    ( format === 'jpg' || format === 'jpeg' ) &&
      s.jpeg({
        'quality'           : 95,
        'chromaSubsampling' : '4:2:2',
      });

    // fill, cover, outside all result in temp files being left behind on Windows 10
    ( width && height ) &&
      s.resize( width, height, { 'fit': 'inside' } );

    return s.toFile( outputPath );

  }
```

Is this a known issue? Can anyone else reproduce? Is there something else I should be doing to avoid it?

**Comment by lovell** [comment]
Hello, I recently fixed this in libvips for Windows users - see https://github.com/libvips/libvips/pull/1155

The next major version of sharp will probably ship with libvips binaries that include the upstream fix for this.

You might be able to workaround it for now by setting the `VIPS_DISC_THRESHOLD` environment variable - see https://libvips.github.io/libvips/API/current/VipsImage.html#vips-image-new-from-file

**Comment by michealhall** [comment]
Just to amend my previous report - last night the temp files started being left behind, even when being used with fit 'inside'.

However, adding the `VIPS_DISC_THRESHOLD=250m` environment var seems to have handled the problem.

Thanks Lovell!

**Comment by lovell** [comment]
sharp v0.22.0 now available, providing libvips v8.7.4.
